src/NServiceBus.Core/Sagas/SagaPersistenceBehavior.cs

```diff
@@ -297,8 +297,14 @@ IContainSagaData CreateNewSagaEntity(SagaMetadata metadata, IInvokeHandlerContex
                     .ConvertFromInvariantString(value.PropertyValue.ToString());
 
                 propertyInfo.SetValue(sagaEntity, convertedValue);
-            }
 
+                sagaEntity.Id = sagaIdGenerator.Generate(value.PropertyName, value.PropertyValue, metadata, context.Extensions);
+            }
+            else
+            {
+                sagaEntity.Id = sagaIdGenerator.Generate(null, null, metadata, context.Extensions);
```

Comment:
I don't really like the approach of passing null values. That will be hard to keep in mind for custom implementations.